internal/graph/resolve_check_request.go

```diff
@@ -0,0 +1,64 @@
+package graph
+
+import (
+	openfgav1 "github.com/openfga/api/proto/openfga/v1"
+	"golang.org/x/exp/maps"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type ResolveCheckRequest struct {
+	StoreID              string
+	AuthorizationModelID string
+	TupleKey             *openfgav1.TupleKey
+	ContextualTuples     []*openfgav1.TupleKey
+	Context              *structpb.Struct
+	RequestMetadata      *ResolveCheckRequestMetadata
+	VisitedPaths         map[string]struct{}
+	Consistency          openfgav1.ConsistencyPreference
+}
+
+func (r *ResolveCheckRequest) clone() *ResolveCheckRequest {
+	return &ResolveCheckRequest{
+		StoreID:              r.StoreID,
+		AuthorizationModelID: r.AuthorizationModelID,
+		TupleKey:             r.TupleKey,
+		ContextualTuples:     r.ContextualTuples,
+		Context:              r.Context,
+		RequestMetadata: &ResolveCheckRequestMetadata{
+			DispatchCounter:     r.GetRequestMetadata().DispatchCounter,
+			Depth:               r.GetRequestMetadata().Depth,
+			DatastoreQueryCount: r.GetRequestMetadata().DatastoreQueryCount,
+			WasThrottled:        r.GetRequestMetadata().WasThrottled,
+		},
+		VisitedPaths: maps.Clone(r.VisitedPaths),
+		Consistency:  r.Consistency,
+	}
+}
+
+func (r *ResolveCheckRequest) GetStoreID() string {
+	return r.StoreID
```

Comment:
Notice I have simplified logic from 

```
func (r *ResolveCheckRequest) GetStoreID() string {
	if r != nil {
		return r.StoreID
	}

	return ""
```

There is no way r == nil and call GetStoreID()

Reply:
How can you prove that `r` is never `nil` in isolation? From the position of this function, you cannot tell how/when/where it will be used.

Reply:
We absolutely _must_ employ defensive programming practices so that we properly handle cases that break our assumptions.

Reply:
Updated

Reply:
I should rephrase my earlier statements to not be so dogmatic:

If there is a sane way to handle a `nil` point receiver value, then we should definitely do it. If the only sane option is to panic, then it may make more sense to do so.

Here, an empty string may make sense when the receiver value is `nil`.